dts/arm/st/mem.h

```diff
@@ -124,6 +124,9 @@
 #elif defined(CONFIG_SOC_STM32L433XC)
 #define DT_FLASH_SIZE		__SIZE_K(256)
 #define DT_SRAM_SIZE		__SIZE_K(64)
+#elif defined(CONFIG_SOC_DUMMY)
```

Comment:
From a quick read, this approach feels kind of ad hoc:

- it's specific to SoC management
- it seems to introduce extra defines we need to test for
- it's got yet another "extract" script which handles certain keys in DTS in magic ways that users won't understand
- it seems to require special magic in a bunch of places to handle the `CONFIG_SOC_DUMMY` step

As discussed on IRC, the only reason we are doing C preprocessor checks in our application DTS overlays is specifically to handle the MCUboot partitions, which I don't think belong in DT in the first place.

There's also this example here of defining memory sizes based on the SoC. But I don't see why those defines can't be generated from the SoC compatible itself in a clean and generic way -- for example, if the YAML binding language was extended so you could generate defines with the right sizes based on which SoC compatible was specified or chosen (not sure if I'm getting the DT lingo right here, but you get what I mean).

So from my perspective, some more generic bindings and solving the MCUboot problem by moving flash partitions to Kconfig defaults would potentially mean there's no need for ifdeffery in the SOC dts overlays at all, and this sort of multi-pass approach wouldn't be needed.

Reply:
> From a quick read, this approach feels kind of ad hoc

I thought that was intended to be ad-hoc, this PR is just to illustrate the discussion we had on IRC. I agree on all the points you list above.

Reply:
> There's also this example here of defining memory sizes based on the SoC. But I don't see why those defines can't be generated from the SoC compatible itself in a clean and generic way -- for example, if the YAML binding language was extended so you could generate defines with the right sizes based on which SoC compatible was specified or chosen (not sure if I'm getting the DT lingo right here, but you get what I mean).

Not sure I follow what you are suggesting.  Where would we encode the memory & flash sizes?  The proper way to handle this is to have different DTS for each SoC variant.

So yes, this is specific to SoC handling for flash & sram sizes. 

We should be avoiding #ifdef in DTS, its not typically the proper way to handle things.

Reply:
> Not sure I follow what you are suggesting. Where would we encode the memory & flash sizes? The proper way to handle this is to have different DTS for each SoC variant.
> 
> So yes, this is specific to SoC handling for flash & sram sizes.
> 
> We should be avoiding #ifdef in DTS, its not typically the proper way to handle things.

I'm confused about the purpose of this RFC if the goal is to get rid of ifdef in DTS, as this introduces additional preprocessor logic.

Reply:
> I'm confused about the purpose of this RFC if the goal is to get rid of ifdef in DTS, as this introduces additional preprocessor logic.

take a look at #9946 to see how I plan on going forward.  The idea was to remove the need for autoconf.h

Reply:
> take a look at #9946 to see how I plan on going forward. The idea was to remove the need for autoconf.h

That one removes some CONFIG_SOC_xxx ifdeffery by adding SoC-specific dtsi files which the board files have to include now, deleting files like dts/arm/ti/mem.h entirely. This PR, on the other hand, adds additional logic to the very same type of mem.h file.

So I don't get how they are related or complementary, sorry; what am I missing? They seem to conflict, actually.

I also don't follow how or if 9946 points out a way forward for users who are doing CONFIG_SOC_xxx ifdeffery in their dts overlays today to do the same thing in a future where DT is run before kconfig.